:seedling: add labels to dockerfile

tuminoid opened the pull request:
Add labels to Dockerfile. `images.version` label will be set by automation during build.

Fix some indentation issues and convert PKGS_LIST to ARG.

Supersedes abandoned PR https://github.com/metal3-io/ironic-image/pull/591

tuminoid commented on Dockerfile:
```diff
@@ -13,19 +13,28 @@ RUN dnf install -y gcc git make xz-devel
 
-ENV PKGS_LIST=main-packages-list.txt
+LABEL org.opencontainers.image.description="Container image to run OpenStack Ironic as part of Metal³"
+LABEL org.opencontainers.image.documentation="https://book.metal3.io/ironic/introduction"
```
Any better documentation urls? I'm OK with this or https://book.metal3.io/ironic/ironic-container-images as well.

elfosardo replied:
https://book.metal3.io/ironic/introduction is good for me

tuminoid commented on Dockerfile:
```diff
@@ -13,19 +13,28 @@ RUN dnf install -y gcc git make xz-devel
+LABEL org.opencontainers.image.url="https://github.com/metal3-io/ironic-image"
+LABEL org.opencontainers.image.vendor="Metal3-io"
+LABEL org.opencontainers.image.version="latest"
+
```
Should we also add authors label? Using what email? 

```text
LABEL org.opencontainers.image.authors="metal3-dev@googlegroups.com"
```
?

tuminoid commented on Dockerfile:
```diff
@@ -13,19 +13,28 @@ RUN dnf install -y gcc git make xz-devel
+LABEL org.opencontainers.image.title="Metal3 Ironic Container"
+LABEL org.opencontainers.image.url="https://github.com/metal3-io/ironic-image"
+LABEL org.opencontainers.image.vendor="Metal3-io"
+LABEL org.opencontainers.image.version="latest"
```
When taking in release automation, we want to update this as part of the build, instead of committing a change to repo. Need to check if the image builder action supports this.

If it doesn't, I rather leave this off, than update this always in a PR before cutting a release.

lentzi90 replied:
Both docker and podman support adding this with `docker build --label ...`. Hopefully we can do the same in the automated builds.
Regarding the CS9 and CS10, how do we handle that currently? Do we have different tags for them? I'm thinking that the version label could be the same as the image tag.

tuminoid replied:
We have two different images in the image-build-action.yml workflow: `ironic`  for CS9 and `ironic_cs10` for CS10 images.

tuminoid replied:
https://github.com/marketplace/actions/docker-build-push-action#inputs supports labels.

I think we should thus leave it off the Dockerfile, and set it during build with 
1. primarily tag if build trigger was tagged (release build)
2. secondarily with the branch name; if branch is main, then latest (push triggered build)

WDYT?

lentzi90 replied:
+1, sounds good!

elfosardo replied:
sounds good to me

tuminoid commented on Dockerfile:
```diff
@@ -13,19 +13,31 @@ RUN dnf install -y gcc git make xz-devel
 FROM $BASE_IMAGE
 
-ENV PKGS_LIST=main-packages-list.txt
+LABEL io.k8s.description="Container image to run OpenStack Ironic as part of Metal³"
```
In BMO we also do `io.k8s.*` labels, do we want to add them too, or take them away from BMO and replace them with org.opencontainers ones?

elfosardo replied:
I would leave org.opencontainers only, unles we have a strong reason to keep io.k8s too

tuminoid replied:
We've had the k8s one in BMO since 2020 by @dhellmann and it is actually also copied now to IRSO by @dtantsur . Googling around it seems to be mostly OpenShift thing: https://docs.redhat.com/en/documentation/openshift_container_platform/4.18/html/images/creating-images#defining-image-metadata

lentzi90 replied:
+1 for skipping the io.k8s labels. If they are needed for openshift I am fine with keeping them though

dtantsur replied:
I don't think we actually care in OpenShift too much. If we discover we need them, we can always re-add in our downstream fork.

elfosardo commented on Dockerfile:
```diff
@@ -13,19 +13,31 @@ RUN dnf install -y gcc git make xz-devel
+LABEL org.opencontainers.image.vendor="Metal3-io"
+LABEL org.opencontainers.image.version="latest"
+
+ARG PKGS_LIST=main-packages-list.txt
```
makes sense, thanks

lentzi90 reviewed (commented):
This looks great!

dtantsur commented:
/approve

I'm okay with skipping the outdated labels

tuminoid commented:
> > /approve
> > I'm okay with skipping the outdated labels
> 
> Thanks. I've dropped the `io.k8s.*` then.
> 
> /hold I'll check the release automation, holding to avoid premature merging. For everything else it seems we have consensus that it is good to go as proposed now.

/unhold
https://github.com/metal3-io/ironic-image/pull/666#discussion_r2030932120 we have agreement to add the `image.version` label during build time via automation. This automation will be implemented in project-infra's common image building action. We do the tag/branch/latest detection there, so we shouldn't need further changes here. I've replaced the `image.version` label with a comment that it will be added build time.

tuminoid commented:
/test metal3-centos-e2e-integration-test-main metal3-ubuntu-e2e-integration-test-main
Just in case.

lentzi90 reviewed (commented):
/lgtm
